tests/integration/test_specfile.py

```diff
@@ -684,8 +679,31 @@ def test_update_version(
     assert spec.expanded_version == version
 
 
-def test_trailing_newline(spec_autosetup, spec_no_trailing_newline):
-    spec = Specfile(spec_autosetup)
+def test_trailing_newline(specfile_factory, spec_autosetup, spec_no_trailing_newline):
+    spec = specfile_factory(spec_autosetup)
     assert str(spec)[-1] == "\n"
-    spec = Specfile(spec_no_trailing_newline)
+    spec = specfile_factory(spec_no_trailing_newline)
     assert str(spec)[-1] != "\n"
+
+
+def test_reload(specfile_factory, spec_minimal, spec_traditional):
+    spec = specfile_factory(spec_minimal)
+    before_reload = copy.deepcopy(spec)
+    if spec.path is not None:
+        shutil.copy2(spec_traditional, spec_minimal)
+        spec.reload()
+        after_reload = spec
+    else:
+        spec._file.seek(0)
+        spec._file.truncate()
+        with open(spec_traditional, encoding="utf-8") as f:
+            new_content = f.read()
+        try:
+            spec._file.write(new_content)
+        except TypeError:
+            spec._file.write(new_content.encode("utf-8"))
```

Comment:
The `errors` argument is important, using `surrogateescape` for both decoding and encoding ensures the original binary representation will be preserved.

```suggestion
        content = Path(spec_traditional).read_text(encoding="utf-8", errors="surrogateescape")
        try:
            spec._file.write(content)
        except TypeError:
            spec._file.write(content.encode("utf-8", errors="surrogateescape"))
```